### PR TITLE
Cookies & Consents Banner: fix broken preview

### LINKED ISSUE #18664: Cookie Widget: Preview from Gutenberg block-based widget screen doesn't display correctly
#### Steps to reproduce the issue

1. Have Jetpack (v9.5-alpha used) active, and make sure extra widgets are turned on in `/wp-admin/admin.php?page=jetpack#/writing`
2. Have the Gutenberg plugin active (v9.8.3 used).
3. In `WP Admin > Appearance > Widgets` add the `Jetpack Cookies & Consents Banner` to a widget area and save changes.
4. Click on `Preview` for the cookie widget (wrapped by the Legacy Widget block wrapper).

#### What I expected

* For a preview of the cookie widget to render.

#### What happened instead

* The cookie widget isn't previewed in a useful way. The widget content is added under the `wp-block-legacy-widget__preview` iframe, though I imagine there is some CSS or rendering conflict at play.

**Screenshots**

Markup on 2021-02-01 at 18:24:32

### LINKED ISSUE #57415: Block Widgets: Cookie & Consents Banner Widget - Widget is not visible in customizer on AT sites
### Quick summary

When adding the Cookies and Consent Banner to the widgets section, you will see that the widget is not visible in the customizer once the settings are saved.

### Steps to reproduce

1. On a WordPress.com site on the business plan, go to Appearance > Customize > Widgets > [Widget Area] > + Add a Widget
2. Add the Cookies and Consents Banner block widget.
3. Widget is not visible after saving the changes and the cookie banner close button on the frontend does not work.

### What you expected to happen

I expect the widget to show in the customizer after saving the changes. And, the banner to close on clicking the close button in the frontend.

### What actually happened

Widget is not visible in the widget section in the customizer although the cookie banner shows on the frontend. The banner does not close on the user's site on clicking the Close button.

### Context

User reported the issue in 4408091-zd-woothemes
I am able to consistently reproduce the issue on AT sites. I tested with Twenty Twenty One, Varia themes.

https://user-images.githubusercontent.com/21078987/139184605-78ec8691-2ae1-4db7-b7f4-0fdf58316e4f.mov

Banner does not close-

https://user-images.githubusercontent.com/21078987/139186139-b1b05d4b-f986-4496-b23c-e0618721c4fb.mov

I am not able to reproduce the banner close issue on my test site. I did the plugin conflict test on the user's site via the Health Check plugin but I still see the banner issue.

### Operating System

_No response_

### Browser

_No response_

### Simple, Atomic or both?

Atomic

### Theme-specific issue?

No

### Other notes

_No response_

### Reproducibility

Consistent

### Severity

Some (< 50%)

### Available workarounds?

No but the platform is still usable

### Workaround details

_No response_

### PR DESCRIPTION
**Pull request by arcangelini**
Fixes Automattic/wp-calypso#57415

#### Changes proposed in this Pull Request:
Due to how the Legacy Widget block loads the preview for this widget it does not show properly. This looks for what page is loading the widget and returns a static version. The check specifically looks at the global `$pagenow` and returns a different output in the event the page is `widgets.php` this accounts for the preview in customizer and widgets.php directly. 

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Go to Appearance ⇢ Customizer ⇢ Widgets
* Add the Legacy Widget block and select the Cookies & Consents Banner
* Adjust the settings and then click off the widget sidebar to view the preview.
* Do the same from Appearance ⇢ Widgets

#### Screenshots
**Customizer**
Markup 2021-12-14 at 09 30 15

**Widgets.php**
Markup 2021-12-14 at 09 30 42

**using Classic Widgets plugin**
Markup 2021-12-14 at 13 49 46

### CODE REVIEW

**Comment by yansern** [comment]
If my understanding is correct from the slack conversation p1628698439384800-slack-CBG1CP4EN, the eu cookie law widget is not meant to be kept in sync with WPCOM side due to code differences, therefore a phab patch for WPCOM has to be manually created.

**Comment by arcangelini** [comment]
> If my understanding is correct from the slack conversation p1628698439384800-slack-CBG1CP4EN, the eu cookie law widget is not meant to be kept in sync with WPCOM side due to code differences, therefore a phab patch for WPCOM has to be manually created.

You are correct @yansern , this does not resolve all the cases in the linked issue. It only resolves the discrepancy on Jetpack. I have not created a diff yet because I am still trying to piece together how that is being called. The changes proposed in this PR do not work at all when applied to WPCOM haha. I will hopefully have an idea of what is going on later today.

In the meantime, we should be able to test and merge this to resolve the issue that JPOP users see.

File: projects/plugins/jetpack/modules/widgets/eu-cookie-law.php [hunk 1]
```diff
@@ -179,8 +179,21 @@ public function widget( $args, $instance ) {
...
+				echo '</div>';
+				return;
+			}
+
 			echo $args['before_widget'];
-			require( dirname( __FILE__ ) . '/eu-cookie-law/widget.php' );
+			require_once __DIR__ . '/eu-cookie-law/widget.php';
 			echo $args['after_widget'];
 			/** This action is already documented in modules/widgets/gravatar-profile.php */
 			do_action( 'jetpack_stats_extra', 'widget_view', 'eu_cookie_law' );
```
Comment (jeherve) [inline projects/plugins/jetpack/modules/widgets/eu-cookie-law.php:199]: Maybe those args and the hook should also be triggered in the widget editor, for consistency?

Comment (arcangelini) [reply projects/plugins/jetpack/modules/widgets/eu-cookie-law.php:199]: The args are what makes it a pop-up. It wraps it in a div that receives all the styling to make it float. That's why I replaced it with the custom div instead.

If there was a way to filter out the args and remove that piece? Or is there another option you think would be better? I tried making styling changes that superceded the position but it never worked.

Comment (jeherve) [reply projects/plugins/jetpack/modules/widgets/eu-cookie-law.php:199]: ah right, I missed that.  Maybe expand on the inline comment you added so we remember all this 6 months from now?

**Review by ivan-ottinger — DISMISSED** [review]
Looks good to me. 👍🏼 Working as described.

It helped to read the [details](https://github.com/Automattic/jetpack/pull/22094/files#r770537020) you shared as the answer to Jeremy's question.

---

Later (in another PR in the future) we could also consider cleaning up the widget settings form to match it closer to what we can see in the "classic" widget editor:

| Classic widget editor | Block editor |
|---|---|
| Markup on 2021-12-17 at 11:23:32 | Markup on 2021-12-17 at 11:24:05 | 

For example as we did with the [Milestone widget](https://github.com/Automattic/jetpack/pull/21302). Unless we already have a working cookie-banner block replacement by that time. :)

File: projects/plugins/jetpack/modules/widgets/eu-cookie-law.php (hunk 1 above)
Comment (arcangelini) [reply projects/plugins/jetpack/modules/widgets/eu-cookie-law.php]: Cool, updated!

**Review by ivan-ottinger — APPROVED** [review]
Looks good to me. 👍🏼

**Review by samiff — APPROVED** [review]

**Pull request merged**